tools/codegen/gen_functionalization_type.py

```diff
@@ -1,16 +1,63 @@
 from tools.codegen.api import cpp
 from tools.codegen.api.types import (
-    DispatcherSignature, Binding, FunctionalizationLambda, ViewInverseSignature
+    DispatcherSignature, Binding, FunctionalizationLambda, ViewInverseSignature,
+    NativeSignature, Expr
 )
 from tools.codegen.api.translate import translate
-from tools.codegen.context import with_native_function
+from tools.codegen.context import (
+    with_native_function, with_native_function_and_selector_and_index,
+    with_native_function_and
+)
 from tools.codegen.model import (
     Argument, NativeFunction, SchemaKind, BackendIndex,
-    Tag, FunctionSchema, SelfArgument, TensorOptionsArguments, BaseType, BaseTy
+    Tag, FunctionSchema, SelfArgument, TensorOptionsArguments, BaseType, BaseTy,
+    NativeFunctionsViewGroup,
 )
 from tools.codegen.selective_build.selector import SelectiveBuilder
 from typing import List, Optional, Union, Tuple
-from tools.codegen.utils import mapMaybe
+
+
+# Generates the body of the default composite C++ kernel for a {view}_copy NativeFunction
+# See Note [view_copy NativeFunctions]
+@with_native_function
+def gen_composite_view_copy_kernel(g: NativeFunctionsViewGroup) -> Optional[str]:
+
+    if g.view_copy is None:
+        return None
+    docstr = f'''
+add_docstr_all("{str(g.view_copy.func.name.name.base)}",
+               r"""
+Performs the same operation as :func:`torch.Tensor.{str(g.view.func.name.name.base)}`, but all output tensors
+are freshly created instead of aliasing the input.
+""")
+'''
+    print(docstr)
+    view_copy_sig = NativeSignature(g.view_copy.func)
+    view_sig = NativeSignature(g.view.func)
```

Comment:
Is Native really the correct signature here? You're calling into ops::call (which is dispatcher signature, IIRC)

Reply:
oof yep, you're right